pipelines/YmlTemplates/deploy.yml

```diff
@@ -0,0 +1,106 @@
+                # Download the 'for-deploy' artifacts
+                - download: ${{ parameters.pipelineDependency }}
+                  artifact: for-deploy
+                  patterns: |
+                    **/for-deploy/**
+                    **/Pipelines/**
+                # Publish NuGet package
+                - task: PowerShell@2
+                  displayName: 'Check nuget count'
+                  inputs:
+                    targetType: 'inline'
+                    script: |
+                      if (Test-Path -Path "$(Pipeline.Workspace)/$(baseRoute)/for-deploy/*.nupkg")
+                      {
+                         exit 0
+                      }
+                      else
+                      {
+                         exit -1
+                      }
+                - task: NuGetCommand@2
+                  displayName: 'NuGet push'
+                  continueOnError: True
+                  inputs:
+                    command: push
+                    packagesToPush: '$(Pipeline.Workspace)/$(baseRoute)/for-deploy/*.nupkg'
+                    nuGetFeedType: ${{ parameters.nuGetFeedType }}
+                    publishFeedCredentials: 'NuGet (Publish)' # Used only for external feed
+                    publishVstsFeed: 'nfield' # Used only for internal feed
+                  condition: succeeded()
+
+                - task: PowerShell@2
+                  displayName: 'Gather Release Info'
+                  inputs:
+                    targetType: 'inline'
+                    script: |
+                      $GitReleaseInfo = Get-Content -Path $(Pipeline.Workspace)/$(baseRoute)/for-deploy/NugetReleaseInfo.txt
+                      Write-Host $GitReleaseInfo
+                      $InfoFields = $GitReleaseInfo.Split(":")
+                      $BranchName = $InfoFields[0]
+                      $ReleaseName = $InfoFields[1]
+                      $CommitHash = $InfoFields[2]
+                      $VersionName = $InfoFields[3]
+                      Write-Host BranchName: $BranchName
+                      Write-Host ReleaseName: $ReleaseName
+                      Write-Host CommitHash: $CommitHash
+                      Write-Host VersionName: $VersionName
+                      Write-Host "##vso[task.setvariable variable=BranchName]$BranchName"
+                      Write-Host "##vso[task.setvariable variable=ReleaseName]$ReleaseName"
+                      Write-Host "##vso[task.setvariable variable=CommitHash]$CommitHash"
+                      Write-Host "##vso[task.setvariable variable=VersionName]$VersionName"
+                      
+                - task: GitHubRelease@1
+                  displayName: 'Create GitHub Release'
+                  condition: ${{ eq(parameters.nuGetFeedType, 'external') }}
+                  inputs:
+                    gitHubConnection: 'github.com_AlbertoBelver' # OAuth Service Conection https://dev.azure.com/niposoftware/Nfield/_settings/adminservices
```

Comment:
This is a public repo. I really don't think that we should write here something like this